core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -609,6 +611,8 @@ protected StructReader(List<ValueReader<?>> readers, Types.StructType struct, Ma
         } else if (field.fieldId() == MetadataColumns.ROW_POSITION.fieldId()) {
           // track where the _pos field is located for setRowPositionSupplier
           this.posField = pos;
+        } else if (field.fieldId() == MetadataColumns.IS_DELETED.fieldId()) {
+          this.readers[pos] = new ConstantReader<>(false);
```

Comment:
Constants are already handled by this class, see the first branch of this if/else logic. I think that it would make more sense to reuse that rather than create a new constant reader.

Reply:
Done.